compiler/rustc_mir_transform/src/coverage/debug.rs

```diff
@@ -434,11 +435,11 @@ impl GraphvizData {
     pub fn get_bcb_coverage_spans_with_counters(
         &self,
         bcb: BasicCoverageBlock,
-    ) -> Option<&Vec<(CoverageSpan, CoverageKind)>> {
+    ) -> Option<&[(CoverageSpan, CoverageKind)]> {
         if let Some(bcb_to_coverage_spans_with_counters) =
             self.some_bcb_to_coverage_spans_with_counters.as_ref()
         {
-            bcb_to_coverage_spans_with_counters.get(&bcb)
+            bcb_to_coverage_spans_with_counters.get(&bcb).map(Deref::deref)
```

Comment:
as_deref works here or no?

Reply:
It doesn't, because `map(deref)` is `Option<&T> -> Option<&T::Target>` and `as_deref` is `&Option<T> -> Option<&T::Target>` (in other words `as_deref = as_ref().map(deref)`). `get` returns `Option<&_>`, so `as_deref` is sadly unusable here.